syncr_backend/network/handle_frontend.py

```diff
@@ -697,3 +703,76 @@ def handle_view_pending_changes(
         }
 
     send_response(conn, response)
+
+
+# Functions for handling incoming frontend requests
+
+
+def handle_request():
+    """
+    Listens for request from frontend and then sends response
+    :return:
+    """
+
+    op_sys = platform.system()
+    if op_sys == 'Windows':
+        _tcp_handle_request()
+    else:
+        _unix_handle_request()
+
+
+def _tcp_handle_request():
+    """
+    Listens for request from frontend and sends response over tcp socket
+    :return:
+    """
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind(FRONTEND_TCP_ADDRESS)
+    s.listen(1)
+    conn, addr = s.accept()
+
+    # Read request from frontend
+    request = b''
+    while True:
+        data = conn.recv(4096)
+        if not data:
+            break
+        else:
+            request += data
+
+    handle_frontend_request(bencode.decode(request), conn)
+
+
+def _unix_handle_request():
+    """
+    Listens for request from frontend and sends response over unix socket
+    :return:
+    """
+
+    try:
+        os.unlink(FRONTEND_UNIX_ADDRESS)
+    except OSError:
+        # does not yet exist, do nothing
+        pass
+
+    s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    s.bind(FRONTEND_UNIX_ADDRESS)
+
+    s.listen(1)
+    conn, addr = s.accept()
+
+    # Read request from frontend
+    request = b''
+    while True:
+        data = conn.recv(4096)
+        if not data:
+            break
+        else:
+            request += data
+
+    handle_frontend_request(bencode.decode(request), conn)
+
+
+if __name__ == '__main__':
+    handle_request()
```

Comment:
this seems to only handle one requests? Shouldn't it be a loop handling requests?

Reply:
youre correct. i just had it as one for testing stuff and forgot to change it. will update in a second.